src/me/vedang/clj_fdb/core.clj

```diff
@@ -146,16 +147,16 @@
                        (partial fimpl/decode (or s k))
                        fimpl/decode)
          keyfn (comp (:keyfn opts identity) key-decoder)
-         valfn (comp (:valfn opts identity) fimpl/decode)]
+         valfn (comp (:valfn opts identity) fimpl/decode)
+         empty-coll (or (empty (:coll opts)) {})]
      (ftr/read tc
                (fn [^Transaction tr]
-                 (reduce (fn [acc ^KeyValue kv]
-                           (assoc acc
-                                  (keyfn (.getKey kv))
-                                  (valfn (.getValue kv))))
-                         {}
-                         (ftr/get-range tr rg)))))))
-
+                 (-> (reduce (fn [acc ^KeyValue kv]
+                               (conj! acc [(keyfn (.getKey kv))
+                                           (valfn (.getValue kv))]))
+                             (transient empty-coll)
+                             (ftr/get-range tr rg))
+                     persistent!))))))
 
```

Comment:
The persistent / transient is a simple enough change (though I find the use of the -> to be overkill here)

Re: The collection option, let me think about how I want to return an ordered range. 

A map makes sense to me since the value cannot and should not be separated from the key. But this does break ordering which is an important property to maintain. I still think that `ordered-map` is the right way to go here.

Reply:
Personally I would always consume (transform or use otherwise in my app) everything that is returned by `get-range` so I find the building up of a hash-map or other map unnecessary. Something like '([k1 v1] [k2 v2] ...)`seems sufficient and if somebody needs a hash-map they can always `(into {} res)`. Especially if you want to write performant code this is just overhead.

I will look at your other comments some time this weekend.